multicloud/src/main/java/de/uniulm/omi/cloudiator/sword/multicloud/pricing/aws/AWSPricingSupplier.java

```diff
@@ -0,0 +1,204 @@
+
+    @Inject
+    public AWSPricingSupplier(@Assisted CloudCredential cloudCredential) {
+        BasicAWSCredentials awsCredentials = new BasicAWSCredentials(cloudCredential.user(), cloudCredential.password());
+        credentialsProvider = new AWSStaticCredentialsProvider(awsCredentials);
+        client = AWSPricingClientBuilder.standard().withRegion("us-east-1").withCredentials(credentialsProvider).build();
+        LOGGER.info("AWSPricing client (with region us-east-1) created");
+    }
+
+    private AWSPricingPack getRawAWSPricingPaginated(String nextToken) {
+        GetProductsRequest getProductsRequest = new GetProductsRequest().withServiceCode("AmazonEC2");
+        GetProductsResult getProductsResult;
+
+        if (nextToken == null || nextToken.isEmpty()){
+            getProductsResult = client.getProducts(getProductsRequest);
+        }
+        else {
+            getProductsResult = client.getProducts(getProductsRequest.withNextToken(nextToken));
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        Set<AWSProductAndTerms> awsProductAndTermsSet = new HashSet<>(100);
+
+        try {
+            for(int i=0; i < getProductsResult.getPriceList().size(); i++) {
+                awsProductAndTermsSet.add(mapper.readValue(getProductsResult.getPriceList().get(i), AWSProductAndTerms.class));
+            }
+        } catch (Exception e) {
+            LOGGER.error("Exception caught while retrieving AWS Price List", e);
+            e.printStackTrace();
+        }
+        return new AWSPricingPack(awsProductAndTermsSet, getProductsResult.getNextToken());
+    }
+
+    @Override
+    public Set<Pricing> get() {
+        String nextToken = null;
+        AWSPricingPack awsPricingPack = null;
+        Set<Pricing> pricings = new HashSet<>();
+
+        int awsPricingPackCounter = 0;
+        LOGGER.info("Starting fetching AWS Pricing Information");
+
+        do {
+            awsPricingPack = getRawAWSPricingPaginated(nextToken);
+
+            /*if(!awsPricingPack.getAwsProductAndTerms().isEmpty()) {
+                LOGGER.info(String.format("AWSPricingPack no: %d", ++awsPricingPackCounter));
+            }
+            else {
+                LOGGER.info("AWSPricingPack is empty");
+            }*/
+
+            for(AWSProductAndTerms awsProductAndTerms : awsPricingPack.getAwsProductAndTerms()) {
+                if(!productFamiliesFilter.contains(awsProductAndTerms.product.productFamily)
+                        || !tenanciesFilter.contains(awsProductAndTerms.product.attributes.get("tenancy"))
+                        || !capacityStatusesFilter.contains(awsProductAndTerms.product.attributes.get("capacitystatus"))
+                        || !awsProductAndTerms.product.attributes.get("operation").startsWith(operationsPatternFilter)) {
+                    continue;
+                }
+
+                OperatingSystemArchitecture operatingSystemArchitecture;
```

Comment:
Choosing OperatingSystemArchitecture could be externalized to separate method.

Reply:
leaving as is